examples/scenarioRoboticArm.py

```diff
@@ -0,0 +1,428 @@
+#  copyright notice and this permission notice appear in all copies.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DIHubLAIMS ALL WARRANTIES
+#  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+#  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+#  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+#  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+#  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+#  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+#
+
+r"""
+Overview
+--------
+
+This scenario demonstrates the capabilities of :ref:`spinningBodyNDOFStateEffector`, which represents an effector with
+N spin axes. The N-degree-of-freedom formulation allows the simulation of any number of attached rigid bodies with
+rotation capable about any axis. The spin axes and mass distribution of the rigid bodies are arbitrary.
+
+The scenario consists of a cylindrical hub with an attached robotic arm. The arm consists of two links that can rotate
+about two different axis. Each axis is controlled through a profiler that moves the arm from the nominal position to a
+final desired angle.
+
+The script is found in the folder ``basilisk/examples`` and executed by using::
+
+    python3 scenarioRoboticArm.py
+
+The scenario outputs four plots: the time history of all angles, of all angle rates, of the hub's attitude and of the
+hub's angular velocity. The scenario also creates a comprehensive Vizard simulation which creates an appropriate
+to-scale model for the defined scenario.
+
+Illustration of Simulation Results
+----------------------------------
+
+.. image:: /_images/Scenarios/scenarioRoboticArm_theta.svg
+   :align: center
+
+.. image:: /_images/Scenarios/scenarioRoboticArm_thetaDot.svg
+   :align: center
+
+.. image:: /_images/Scenarios/scenarioRoboticArm_sigma_BN.svg
+   :align: center
+
+.. image:: /_images/Scenarios/scenarioRoboticArm_omega_BN_B.svg
+   :align: center
+
+"""
+
+#
+# Basilisk Scenario Script and Integrated Test
+#
+# Purpose:  Illustrates the different simulation capabilities of the NDOF modules
+# Author:   João Vaz Carneiro
+# Creation Date:  May 15, 2024
+#
+
+import os
+import matplotlib.pyplot as plt
+import numpy as np
+
+from Basilisk.utilities import (SimulationBaseClass, vizSupport, simIncludeGravBody, macros, orbitalMotion, unitTestSupport)
+from Basilisk.simulation import spacecraft, spinningBodyNDOFStateEffector, prescribedRotation1DOF
+from Basilisk.architecture import messaging
+
+from Basilisk import __path__
+
+bskPath = __path__[0]
+fileName = os.path.basename(os.path.splitext(__file__)[0])
+
+
+def run(show_plots):
+    scSim = createSimBaseClass()
+
+    scGeometry = geometryClass()
+    scSim, scObject = setUpSpacecraft(scSim, scGeometry)
+    scSim, scObject, roboticArmEffector, _ = setUpEffector(scSim, scObject, scGeometry)
```

Comment:
These messages are being returned because they need to be moved to the global function scope. Otherwise, the message objects will be garbage collected, leading to a segmentation fault at run time. The alternative to this would be to save the message object inside ``scSim``, which is already being returned. Would that be preferable?

Reply:
Ah, I see what you are doing.  The other trick Patrick showed me a while back is to use the `disown()` method.  If you search our python scripts, you'll see it used for this reason.  

To make the justification for this clearer to a new user, could you please add a quick comment on why msgs are being returned and not used?  Otherwise new students will have to rediscover the reason you did what you did.  Or, you can try to use `disown()` inside the method that create the message.

Reply:
Using ``disown()`` might be the best option. Otherwise, I'll add a comment explaining this reasoning. I'll see what works best.